CoreScriptingView/src/au/gov/asd/tac/constellation/views/scripting/graph/SGraph.java

```diff
@@ -56,14 +56,14 @@
     public static final GraphElementType LINK = GraphElementType.LINK;
 
     private ScriptEngine engine;
-    private final Graph graph;
+    private final Graph graphObject;
 
     private final ArrayList<SReadableGraph> readableGraphs;
     private final ArrayList<SWritableGraph> writableGraphs;
 
     public SGraph(final ScriptEngine engine, final Graph graph) {
         this.engine = engine;
-        this.graph = graph;
+        this.graphObject = graph;
```

Comment:
Is this sort of thing really an issue? I thought it was fine given the variables have clearly defined, different scopes.

Reply:
Yeah I'm not really sure. @arcturus2 do you think that its worth changing the variable names?

Reply:
I think the issue is that it can confuse users of the class. Yes their definitions are clearly defined but unless you happen to know that there are 2 different fields with the same name in the same class, there may be some confusion down the line when you use the wrong one and things aren't happening you expect them to.

Really, this is just poor coding practice. Hence this is a code smell and not something like a bug

Reply:
Generally I'm a fan of following SonarQube though this particular scenario does seem a little overkill. I think though given the definition of `GRAPH` and `graph` is isolated to just this class it is worth resolving the code smell to avoid the chance someone does get confused. I think what this SQ rule is trying to address is important so happy to support it at the end of the day.

As always, nothing is set in stone for me so if there are any objections then happy to revisit it.